### DISCUSSION TITLE
How do I add two lenses to a model?

### DISCUSSION THREAD

**Discussion by mccoyn** [mjhoptics/ray-optics#82] [Q&A] [answered]
I would like to ray trace using two lens elements from the zemax files.  I can't figure out how to do it.  If I follow the "ThorLabs catalog lens example" it appears to make a model from a single lens.  If I follow the "Triplet example", the lenses are specified directly as surfaces rather than using a zemax file.

I want to combine these two.  Load two zemax files and combine them into one model.  How can I do this?

**Comment by mjhoptics** [comment, marked as answer]
There is a jupyter notebook repo called ray-optics-notebooks that has an example of what you're trying to do. It's called [AddFromFile.ipynb](https://github.com/mjhoptics/ray-optics-notebooks/blob/master/AddFromFile.ipynb). Let me know if you have further questions. Thanks, Mike

**Comment by mccoyn** [comment]
Thanks.  That looks like a good set of examples.  In my case, I think it wasn't able to get it to work because I left out the "sequences.gaps[0].thi = 1e10" line.
